### PR TITLE
CHECKOUT-1156: add button/link states, inherit default styles

### PR DESCRIPTION
**Pull request by jordanwink201**
Use stencil theme default properties by default
- add button states for `primary`, `tertiary`, and `action` button types
- add link states
- add body and header color
- organize checkout sass into stencil folder/file structure

screen shot 2016-10-21 at 2 35 53 pm

@icatalina @davidchin @bigcommerce/checkout @bc-miko-ademagic @mcampa

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -37,23 +37,51 @@
...
-    "checkout_header_backgroundColor": "#fafafa",
-    "checkout_main_backgroundColor": "#ffffff",
-    "checkout_cart_backgroundColor": "#ffffff",
-    "checkout_headings_font": "Google_Montserrat_400",
-    "checkout_body_font": "Google_Karla_400",
-    "checkout_body_color": "#f010da",
-    "checkout_links_color": "#f010da",
-    "checkout_buttons_backgroundColor": "#454545",
-    "checkout_errors_color": "#f010da",
-    "checkout_formField_backgroundColor": "white",
```
Comment (mcampa) [inline config.json:50]: Please consider that these variables are already used by merchants. Renaming or changing a value could potentially break merchant themes

Comment (mcampa) [reply config.json:50]: If they are only used in UCO, should be fine

Comment (jordanwink201) [reply config.json:50]: These will be renamed to `optimized-checkout` to be more unique and consistent with PR #7

**Comment by jordanwink201** [comment]
Due to conflicting inherited styles ex: `default link color` (Edit Cart & Promo/Gift Certificate links), will not inherit any styles from storefront for initial release.

screen shot 2016-10-24 at 4 13 55 pm

**Pull request closed without merging**